pkgs/development/compilers/zig/0.10.nix

```diff
@@ -60,9 +60,7 @@ stdenv.mkDerivation (finalAttrs: {
     "-DZIG_TARGET_MCPU=baseline"
   ];
 
-  preBuild = ''
-    export HOME=$TMPDIR;
-  '';
+  env.ZIG_GLOBAL_CACHE_DIR = "$NIX_BUILD_TOP/zig-cache";
```

Comment:
what does NIIX_BUILD_TOP do? I cannot find any documentation for it.

Reply:
It is an obscure environment variable, documented in Nix the evaluator:

https://github.com/NixOS/nix/blob/84050709ea18f3285a85d729f40c8f8eddf5008e/doc/manual/src/language/derivations.md?plain=1#L105

Basically it points to the root of where build will happen.

(However I will change this to the more typical TMPDIR).